src/poly.rs

```diff
@@ -128,6 +131,50 @@ impl<F, B> Polynomial<F, B> {
     }
 }
 
+pub(crate) fn batch_invert_assigned<F: FieldExt>(
+    assigned: &[Polynomial<Assigned<F>, LagrangeCoeff>],
+) -> Vec<Polynomial<F, LagrangeCoeff>> {
+    let mut assigned_denominators: Vec<_> = assigned
+        .iter()
+        .map(|f| {
+            f.iter()
+                .map(|value| value.denominator())
+                .collect::<Vec<_>>()
+        })
+        .collect();
+
+    assigned_denominators
+        .iter_mut()
+        .flat_map(|f| {
+            f.iter_mut()
+                // If the denominator is zero or one, we can skip it, reducing the
+                // size of the batch inversion.
```

Comment:
```suggestion
                // size (and for zeros, ensuring correctness) of the batch inversion.
```

Reply:
I believe this comment is unnecessary, because `.batch_invert()` leaves zeros alone.

Reply:
Yes, I had misremembered the contract of `batch_invert`.